src/view/surface.cpp

```diff
@@ -164,7 +164,8 @@ void wf::surface_interface_t::impl::scale_opaque_region(
      * different scales, we just shrink by 1 to compensate for the ceil/floor
      * discrepancy */
     int ceil_factor = 0;
-    if (output->handle->scale != (float)wsurface->current.scale)
+    if ((wsurface && output->handle->scale != (float)wsurface->current.scale) ||
+        (!wsurface && output->handle->scale != std::round(output->handle->scale)))
         ceil_factor = 1;
```

Comment:
style: since this is a multiline if, it needs parenthesis.
Tbh I'm getting more and more inclined towards saying we should always use parenthesis so that the rule is simple

Reply:
> style: since this is a multiline if, it needs parenthesis.
> Tbh I'm getting more and more inclined towards saying we should always use parenthesis so that the rule is simple

My personal rule is to only use braces when necessary. This basically means use braces unless it's a conditional with only one LOC in the body. It seems strange to me that braces would be required depending on how many lines the conditional happens to be. I think it would make more sense to require braces always since you seem to be quite pedantic about it.